### PR TITLE
Add distorted frame to shape time dependence

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

This is in preparation for using this time dependence with the shape control system.

Also
- Adds a size function of time in the shape time dependence. This function of time doesn't control any maps, but is necessary if we want to use this time dependence with the shape control system (because the shape control error requires there to be a size function of time)
- Templates the shape time dependence on an `ObjectLabel` so we know which function of time name to use.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/Creators/TimeDependence/Shape.hpp [hunk 1]
```diff
@@ -53,14 +54,32 @@ namespace domain::creators::time_dependence {
  * by the value of `inner_radius`. If the user wants the inner surface of the
  * Shape to conform to a Kerr horizon for a given mass and spin, `inner_radius`
  * should be the Boyer-Lindquist radius of the outer horizon.
+ *
+ * The actual shape map that is applied will go from the Grid to the Distorted
+ * frame, and then an identity map will go from the Distorted to Inertial frame.
+ * The reasoning behind this is because in basically all use cases (BBH), the
+ * shape map will go to the Distorted frame only and other maps will go to from
```
Comment (nilsvu) [inline src/Domain/Creators/TimeDependence/Shape.hpp]: ... go <s>to</s> from ...

File: src/Domain/Creators/TimeDependence/Shape.hpp [hunk 2]
```diff
@@ -53,14 +54,32 @@ namespace domain::creators::time_dependence {
...
+ * the Distorted frame to the Inertial frame.
  */
 class Shape final : public TimeDependence<3> {
  private:
   using ShapeMap = domain::CoordinateMaps::TimeDependent::Shape;
+  using Identity = domain::CoordinateMaps::Identity<3>;
+  using GridToInertialMap =
+      detail::generate_coordinate_map_t<Frame::Grid, Frame::Inertial,
+                                        tmpl::list<ShapeMap, Identity>>;
```
Comment (nilsvu) [inline src/Domain/Creators/TimeDependence/Shape.hpp]: delete `Identity` here? That should still work I think.

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Review by wthrowe — DISMISSED** [review]
I have no additional comments.

**Comment by knelli2** [comment]
(hopefully) Fixed a linking error on macOS.

**Comment by knelli2** [comment]
After a discussion at the SpECTRE call on Thurs Feb. 16 2023, it was decided that we needed to fix the cyclic dependency of the TimeDependence library and move the ObjectLabel to DomainStructure. Both in separate PRs.

While I was at I also decided to move the renaming of excision spheres to another PR since it can be a stand-alone change.

Now this PR depends on and includes #4747, #4746, #4745.

**Comment by knelli2** [comment]
No longer dependent on #4747, #4746, #4745.

**Review by nilsvu — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**